Sources/TokamakCore/Shapes/Path.swift

```diff
@@ -0,0 +1,291 @@
+// Copyright 2020 Tokamak contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//  Created by Carson Katri on 06/28/2020.
+//
+
+/// The outline of a 2D shape.
+public struct Path: Equatable, LosslessStringConvertible {
+  public var description: String {
+    """
+    \(storage)
+    \(elements)
+    \(transform)
+    """
+  }
+
+  public enum Storage: Equatable {
+    case empty
+    case rect(CGRect)
+    case ellipse(CGRect)
+    indirect case roundedRect(FixedRoundedRect)
+    indirect case stroked(StrokedPath)
+    indirect case trimmed(TrimmedPath)
+//    case path(PathBox)
+  }
+
+  public enum Element: Equatable {
+    case move(to: CGPoint)
+    case line(to: CGPoint)
+    case quadCurve(to: CGPoint, control: CGPoint)
+    case curve(to: CGPoint, control1: CGPoint, control2: CGPoint)
+    case closeSubpath
+  }
+
+  public var storage: Storage
+  public var elements: [Element] = []
+  public var transform: CGAffineTransform = .identity
+
+  public struct _SubPath: Equatable {
+    let path: Path
+    let transform: CGAffineTransform
+  }
+
+  public var subpaths: [_SubPath] = []
+
+  public init() {
+    storage = .empty
+  }
+
+  init(storage: Storage) {
+    self.storage = storage
+  }
+
+  public init(_ rect: CGRect) {
+    storage = .rect(rect)
+  }
+
+  public init(roundedRect rect: CGRect,
+              cornerSize: CGSize,
+              style: RoundedCornerStyle = .circular) {
+    storage = .roundedRect(FixedRoundedRect(rect: rect,
+                                            cornerRadius: cornerSize.width,
+                                            style: style))
+    // FIXME: This currently doesn't support an actualy CGSize. I'm not sure how that works or what it does.
```

Comment:
Looks like if you do `.clipShape(RoundedRectangle(cornerSize: CGSize(width: 20, height: 10), style: .circular))` you get this:
<img width="225" alt="Screenshot_2020-06-29 12 33 11" src="https://user-images.githubusercontent.com/25517624/86032030-b6ef6800-ba04-11ea-813e-56ba18820e98.png">

If you use `.continuous`, it looks like it just uses the `width` value.

Reply:
Thanks for the reference! I implemented it to match.